front-end/scripts/main.js

```diff
@@ -0,0 +1,418 @@
+                document.querySelector('#global-icicle-carriage').classList.add('removed');
+            } else {
+                document.querySelector('#global-icicle-all').classList.add('removed');
+                document.querySelector('#global-icicle-disease').classList.add('removed');
+                document.querySelector('#global-icicle-carriage').classList.remove('removed');
+            }
+        });
+    });
+    
+    
+    // Show and close country view modal 
+    map.addEventListener('click', showModal);
+    closeBtn.addEventListener('click', closeModal);
+    overlay.addEventListener('click', closeModal);
+    
+    function showModal(e) {
+        let selectedCountry = getCountry(e.target);
+
+        // Limit response to countries with available data
+        if (countries.indexOf(selectedCountry) !== -1) {
+            let countryViewTitle = document.querySelector('#country-view-title');
+            countryViewTitle.innerHTML = `<h1>${alpha2[selectedCountry]}</h1>`;
+            overlay.classList.remove('removed');
+            modal.classList.remove('removed');
+
+            // Display suitable age group information based on data.json
+            let ageGroups = summary[selectedCountry]['ageGroups'];
+            let selectedAgeGroup = 0;
+            let selectedType = 'all';
+
+            if (ageGroups.filter(Boolean).length === 2) {
+                ageBothDiv.classList.remove('removed');
+                age0Div.classList.add('removed');
+                age1Div.classList.add('removed');
+            } else if (ageGroups[0] === true) {
+                ageBothDiv.classList.add('removed');
+                age0Div.classList.remove('removed');
+                age1Div.classList.add('removed');
+            } else {
+                ageBothDiv.classList.add('removed');
+                age0Div.classList.add('removed');
+                age1Div.classList.remove('removed');
+                selectedAgeGroup = 1;
+            }
+
+            // Draw the sunburst charts and (removed) bar charts with default parameters
+            let periods = summary[selectedCountry]['periods'];
+            sunburst(selectedCountry, selectedType, selectedAgeGroup, periods, data['country'][selectedCountry][selectedType][`age${selectedAgeGroup}`], data["domainRange"]);
+            barchart(data['country'][selectedCountry]['resistance'][`age${selectedAgeGroup}`], data['antibiotics']);
+
+            // addEventListener to country view age and type selectors
+            // Update country view charts based on current selectors
+            document.querySelectorAll('input[name="country-view-age"]').forEach(input => {
+                input.addEventListener('change', (e) => {
+                    selectedAgeGroup = e.target.value;
+                    sunburst(selectedCountry, selectedType, selectedAgeGroup, periods, data['country'][selectedCountry][selectedType][`age${selectedAgeGroup}`], data["domainRange"]);
```

Comment:
I've noticed that when I change the age group or type, it resets. I think for a user it might be more intuitive to keep the current serotype and GPSC selection when changing age group/type